Client/Networking/NetworkSession.swift

```diff
@@ -23,6 +30,40 @@ extension URLSession: NetworkSession {
       httpVersion: "HTTP/1.1",
       headerFields: [:])!
   }
+  
+  func dataRequest(with url: URL, _ completion: @escaping (Result<NetworkSessionDataResponse, Error>) -> Void) {
+    self.dataTask(with: url) { data, response, error in
+      if let error = error {
+        completion(.failure(error))
+      } else {
+        completion(.success((data ?? Data(), response ?? self.createResponse(url: url))))
```

Comment:
should we return a failure if `data` is nil?

Reply:
I left it like this because the previous code was doing it, but I think it's actually better to return an error if it is nil. Though some requests might not return any data for the body.